tests/worker_deployment_test.go

```diff
@@ -0,0 +1,148 @@
+		dynamicconfig.FrontendNamespaceReplicationInducingAPIsRPS.Key():                               1000,
+	}))
+}
+
+func (s *WorkerDeploymentSuite) SetupTest() {
+	s.FunctionalTestSuite.SetupTest()
+}
+
+// pollFromDeployment calls PollWorkflowTaskQueue to start deployment related workflows
+func (s *WorkerDeploymentSuite) pollFromDeployment(ctx context.Context, taskQueue *taskqueuepb.TaskQueue,
+	deploymentName string, version string) {
+	tv := testvars.New(s)
+
+	_, _ = s.FrontendClient().PollWorkflowTaskQueue(ctx, &workflowservice.PollWorkflowTaskQueueRequest{
+		Namespace: s.Namespace().String(),
+		TaskQueue: taskQueue,
+		Identity:  tv.Any().String(),
+		WorkerVersionCapabilities: &commonpb.WorkerVersionCapabilities{
+			UseVersioning:        true,
+			BuildId:              version,
+			DeploymentSeriesName: deploymentName,
+		},
+	})
+}
+
+func (s *WorkerDeploymentSuite) TestDescribeWorkerDeployment() {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*15)
+	defer cancel()
+	tv := testvars.New(s)
+
+	// Starting two versions of the deployment
+	firstVersion := tv.WithDeploymentVersionNumber(1)
+	secondVersion := tv.WithDeploymentVersionNumber(2)
+
+	go s.pollFromDeployment(ctx, tv.TaskQueue(), tv.DeploymentName(), firstVersion.DeploymentVersion())
+	go s.pollFromDeployment(ctx, tv.TaskQueue(), tv.DeploymentName(), secondVersion.DeploymentVersion())
+
+	s.EventuallyWithT(func(t *assert.CollectT) {
+		a := assert.New(t)
+
+		resp, err := s.FrontendClient().DescribeWorkerDeployment(ctx, &workflowservice.DescribeWorkerDeploymentRequest{
+			Namespace:      s.Namespace().String(),
+			DeploymentName: tv.DeploymentName(),
+		})
+		a.NoError(err)
+		a.NotNil(resp.GetWorkerDeploymentInfo())
+		a.Equal(tv.DeploymentName(), resp.GetWorkerDeploymentInfo().GetName())
+
+		a.NotNil(resp.GetWorkerDeploymentInfo().GetVersionSummaries())
+		a.Equal(2, len(resp.GetWorkerDeploymentInfo().GetVersionSummaries()))
+
+		if len(resp.GetWorkerDeploymentInfo().GetVersionSummaries()) < 2 {
+			return
+		}
+		a.NotNil(resp.GetWorkerDeploymentInfo().GetVersionSummaries()[0].GetVersion())
+		a.NotNil(resp.GetWorkerDeploymentInfo().GetVersionSummaries()[1].GetVersion())
+
+		versions := []string{
+			resp.GetWorkerDeploymentInfo().GetVersionSummaries()[0].GetVersion(),
+			resp.GetWorkerDeploymentInfo().GetVersionSummaries()[1].GetVersion(),
+		}
+		a.Contains(versions, firstVersion.DeploymentVersion())
+		a.Contains(versions, secondVersion.DeploymentVersion())
+
+		a.NotNil(resp.GetWorkerDeploymentInfo().GetVersionSummaries()[0].GetCreateTime())
+		a.NotNil(resp.GetWorkerDeploymentInfo().GetVersionSummaries()[1].GetCreateTime())
+
+		a.NotNil(resp.GetWorkerDeploymentInfo().GetCreateTime())
+	}, time.Second*10, time.Millisecond*1000)
+}
+
+// Name is used by testvars. We use a shorten test name in variables so that physical task queue IDs
+// do not grow larger that DB column limit (currently as low as 272 chars).
```

Comment:
```suggestion
// Name is used by testvars. We use a shortened test name in variables so that physical task queue IDs
// do not grow larger than DB column limit (currently as low as 272 chars).
```